mobile-app/app/screens/TransactionAuthorization/PasscodePrompt.tsx

```diff
@@ -33,157 +36,131 @@ interface PasscodePromptProps {
-                    dark={tailwind('text-gray-400')}
-                    light={tailwind('text-gray-500')}
-                    style={tailwind('p-2 text-sm text-center mb-4')}
-                  >
-                    {props.additionalMessageUrl}
-                  </ThemedText>
-                </View>
-              )}
-
-              <View style={tailwind('px-8 text-sm text-center mb-6')}>
-                <ThemedText
+        <ThemedTextV2
+          style={tailwind('text-center font-normal-v2 pt-1.5 px-10', { 'mb-16': props.status !== TransactionStatus.SIGNING && props.status !== TransactionStatus.AUTHORIZED })}
+        >
+          {props.title}
+        </ThemedTextV2>
+        {props.status === TransactionStatus.SIGNING && <ThemedActivityIndicatorV2 style={tailwind('py-2 my-5')} />}
+        {props.status === TransactionStatus.AUTHORIZED && <SuccessIndicator />}
+        <PinTextInputV2
+          cellCount={props.pinLength}
+          onChange={(pin) => {
+            props.onPinInput(pin)
+          }}
+          testID='pin_authorize'
+          value={props.pin}
+          style={props.status !== TransactionStatus.SIGNING && props.status !== TransactionStatus.AUTHORIZED ? 'mt-3' : ''}
+        />
+        <View style={tailwind('text-sm text-center mb-14 mt-4 px-10')}>
+          {// show pin success message
+              props.status === TransactionStatus.AUTHORIZED &&
+                <ThemedTextV2
                   testID='txn_authorization_message'
-                  dark={tailwind('text-gray-400')}
-                  light={tailwind('text-gray-500')}
-                  style={tailwind('p-2 px-8 text-sm text-center mb-2')}
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('text-sm text-center font-normal-v2')}
                 >
-                  {props.message}
-                </ThemedText>
-
-                {
-                  props.transaction?.description !== undefined && (
-                    <ThemedText
-                      testID='txn_authorization_description'
-                      dark={tailwind('text-gray-400')}
-                      light={tailwind('text-gray-500')}
-                      style={tailwind('text-sm text-center')}
-                    >
-                      {props.transaction.description}
-                    </ThemedText>
-                  )
-                }
-              </View>
-
-              {
-                props.status === TransactionStatus.PIN && (
-                  <PinTextInput
-                    cellCount={props.pinLength}
-                    onChange={(pin) => {
-                      props.onPinInput(pin)
-                    }}
-                    testID='pin_authorize'
-                    value={props.pin}
-                  />
+                  {props.successMessage}
+                </ThemedTextV2>
+            }
+          {// show loading message
```

Comment:
Can we abstract this into one component? We have the same component, just different states and messages. Code is a bit messy right now.

Reply:
@thedoublejay , there a reason behind that. If we abstract it and make it into one component, there will be unnecessary spacing when there is a invalid password message.

Reply:
What do you mean? Can't it be fixed with styles?

Reply:
@thedoublejay, refactored. Let me know how you think.